CDCgov/prime-simplereport
# set user account creation fields required

# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5881 

## Changes Proposed

- Set required = true for fields
- Replace userEvent with fireEvent in test that was updated

## Additional Information

- decisions that were made
- notice of future work that needs to be done

## Testing

1. Create a new account in dev7
2. copy the link address in the verification email 
3. Replace `test.simplereport.gov` with `dev7.simplereport.gov`
4. Visit these all of the forms and confirm they have the required attribute and `*`
- /uac/set-password
- /uac/set-recovery-question
- /uac/mfa-select
- /uac/mfa-okta/verify
- /uac/mfa-google-auth/verify
- /uac/mfa-phone/verify
- /uac/mfa-sms
- /uac/mfa-sms/verify
- /uac/mfa-phone
- /uac/mfa-phone/verify
- /uac/mfa-email/verify
## Screenshots / Demos

- Set password 
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/d799c659-1f09-410c-a62a-ddcd7fe0d536)
- Set recovery question
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/88112498-8318-4f65-8e94-b50e1cec0073)
- MFA Select
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/36c349b1-1b3f-43e7-9fd3-7142cee10411)
- Okta verify
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/17cdeb82-140a-412f-a4fb-09ac2be6ad9c)
- Google Auth verify
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/4169c140-9ddc-4631-b682-6ce2653be3c9)
- Phone
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/b629e591-aa5b-4660-ba21-cf784b8c1880)
- Phone verify
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/c7e8e52f-78fe-4a60-ace7-dd287af55a41)
- SMS
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/d5fc4bce-ac78-4801-a0f6-c790e29b10d4)
- SMS Verify
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/77aaa941-acb3-4ad2-9768-a45d10858acb)
- Email Verify
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/5f4fb7a8-cc89-4c1f-9a78-51136825fe42)
